### ISSUE TITLE
Unexpected NaN when raising `x + 0j` to infinity

### ISSUE THREAD

**Issue by lfaucheux** [google/jax#17343]
### Description

I am getting unexpected nan with the following example.

```python
# >>> import numpy as onp; onp.power(1 + 0j, onp.inf)
# (1+0j)
>>> import jax.numpy as jnp; jnp.power(1 + 0j, jnp.inf)
Array(nan+nanj, dtype=complex128, weak_type=True)
```

### What jax/jaxlib version are you using?

0.4.13

### Which accelerator(s) are you using?

CPU

### Additional system info

Python 3.8

**Comment by jakevdp** [comment]
Thanks for the report – it looks like this comes from the native XLA power implementation, rather than having to do with anything in the `jax.numpy` layer:
```python
>>> from jax import lax
>>> print(lax.pow(1 + 0j, float('inf') + 0j))
(nan+nanj)
```
We should report this at https://github.com/openxla/xla

**Comment by jakevdp** [comment]
Which version of numpy are you using? In NumPy 1.25.2, I am also getting NaNs for the NumPy version of the computation.

**Comment by jakevdp** [comment]
Reported upstream in https://github.com/openxla/xla/issues/5369

**Comment by lfaucheux** [comment]
> Which version of numpy are you using? In NumPy 1.25.2, I am also getting NaNs for the NumPy version of the computation.

1.24.3

**Comment by jakevdp** [comment]
I see NaN outputs on numpy 1.24.3 as well:
```python
import numpy as np
print(np.__version__)
print(np.power(1 + 0j, np.inf))
```
```
1.24.3
(nan+nanj)
<~>:3: RuntimeWarning: invalid value encountered in power
  print(np.power(1 + 0j, np.inf))
```
The behavior might be platform-dependent.

**Comment by lfaucheux** [comment]
True.

![osdep](https://github.com/google/jax/assets/19575612/cda5807f-921a-4e1d-aabf-b3c2e5f63746)

**Comment by lfaucheux** [comment]
True

![osdep](https://github.com/google/jax/assets/19575612/29730176-5f82-40a2-b91d-b6eaf9e3f567)

**Comment by akuegel** [comment]
C++ std::power also gives that result. But yes, I guess we can make that a special case that if base is 1, result will be 1.

**Comment by jakevdp** [comment]
This should be fixed as of JAX v0.4.16
